test/forkMainnet/ConditionalSwap/Fill.t.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { IConditionalSwap } from "contracts/interfaces/IConditionalSwap.sol";
+import { ConditionalOrderSwapTest } from "test/forkMainnet/ConditionalSwap/Setup.t.sol";
+import { BalanceSnapshot, Snapshot } from "test/utils/BalanceSnapshot.sol";
+
+contract ConFillTest is ConditionalOrderSwapTest {
+    using BalanceSnapshot for Snapshot;
+
+    function setUp() public override {
+        super.setUp();
+    }
+
+    function testBestBuyOrder() external {
```

Comment:
add comments for different scenario requirement and flags to help reader understanding